### PR TITLE
Initial version of classes spin and spin_system

### PR DESCRIPTION
**Pull request by ak-ustutt**
I have added two classes:

class spin  holds a spin with its properties (like main axes, g values, ZFS, both electron and nuclear spins are allowed)

class spin_system  holds a collection of spins and allows to define interactions; it returns the corresponding spin, magnetic moment and Hamiltonian matrices

proper documentation is still missing, but a unit test for some basic functionalities is provided

### CODE REVIEW

**Comment by ak-ustutt** [comment]
Just for quick review and for your attention; do not spent too much time on it

File: packages/koehnlab/spin_hamiltonians/spin_systems.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
+import numpy as np 
+from .phys_const import ge, muNbohr
+from .spin_utils import spinMat
+
+
+numThr = 1e-8
+maxDim = 1000 
+
+class spin:
```
Comment (Krzmbrzl) [inline packages/koehnlab/spin_hamiltonians/spin_systems.py:9]: Classnames should start with a capital letter in order to make it easy to distinguish class constructions from function calls.

File: packages/koehnlab/spin_hamiltonians/spin_systems.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+    def __init__(self,S,type="e",gnuc=None):
```
Comment (Krzmbrzl) [inline packages/koehnlab/spin_hamiltonians/spin_systems.py:11]: `type` is superfluous. You could simply check if `gnuc` is `None` or not. If you want to keep `type` for possible future extensions, creating an `enum` for the possible values will greatly improve usability of the function (the set of allowed values is immediately clear and typos can be caught by IDEs and not only at runtime)

File: packages/koehnlab/spin_hamiltonians/spin_systems.py [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+            self.ZFaxes = self.axes
+        self.ZFaxial = ZFaxial
+        self.ZFrhombic = ZFrhombic
+     
+        self.ZFten = np.matmul(self.ZFaxes,
+                     np.matmul(np.diag([-self.ZFaxial/3.+self.ZFrhombic,-self.ZFaxial/3.-self.ZFrhombic,2*self.ZFaxial/3.]),
+                               self.ZFaxes.T))
+
+
+    def getSpinMat(self):
```
Comment (Krzmbrzl) [inline packages/koehnlab/spin_hamiltonians/spin_systems.py:69]: Inconsistent naming of functions - other functions use snake_case whereas here you use camelCase.

**Review by Krzmbrzl — COMMENTED** [review]
Comments on `spin` class also apply to `spin_system`.

Doc-strings for functions might be useful as well to ensure everyone is on the same page for what a given function does

File: packages/koehnlab/spin_hamiltonians/spin_systems.py (hunk 1 above)
Comment (ak-ustutt) [reply packages/koehnlab/spin_hamiltonians/spin_systems.py:9]: Ah OK, I am learning :) ... can you apply the change?

File: packages/koehnlab/spin_hamiltonians/spin_systems.py (hunk 2 above)
Comment (ak-ustutt) [reply packages/koehnlab/spin_hamiltonians/spin_systems.py:11]: I thought that giving the type is a bit more explicit; actually, I was originally thinking of giving the isotope here to set gnuc automatically. I guess this feature is still in the flow ... I did not look up how to implement an enum yet, can you add this quickly?

File: packages/koehnlab/spin_hamiltonians/spin_systems.py (hunk 3 above)
Comment (ak-ustutt) [reply packages/koehnlab/spin_hamiltonians/spin_systems.py:69]: I am probably very inconsistent at times ... I have to have a look at all the other functions, do not remember ...

Comment (ak-ustutt) [reply packages/koehnlab/spin_hamiltonians/spin_systems.py:69]: OK, I will probably introduce snake consistently. I could do the renaming of the class then as well.

**Pull request merged**